runtime/lua/vim/lsp.lua

```diff
@@ -890,6 +890,42 @@ function lsp.start_client(config)
     end)
   end
 
+  --@private
+  --- Sends a request to the server and synchronously waits for the response.
+  ---
+  --- This is a wrapper around {client.request}
+  ---
+  --@param method (string) LSP method name.
+  --@param params (table) LSP request params.
+  --@param timeout_ms (number, optional, default=100) Maximum time in
+  ---milliseconds to wait for a result.
+  --@param bufnr (number) Buffer handle (0 for current).
+  --@returns { err, result }, where `err` and `result` come from the |lsp-handler|.
+  ---On timeout, cancel or error, returns `(nil, err)` where `err` is a
+  ---string describing the failure reason. If the request was unsuccessful
+  ---returns `nil`.
+  --@see |vim.lsp.buf_request_sync()|
+  function client.request_sync(method, params, timeout_ms, bufnr)
+    local request_result = nil
+    local function _sync_handler(err, _, result)
+      request_result = { error = err, result = result }
+    end
+
+    local success, request_id = client.request(method, params, _sync_handler,
+      bufnr)
+    if not success then return nil end
+
+    local wait_result, reason = vim.wait(timeout_ms or 100, function()
```

Comment:
Is the timeout_ms high enough for prettier and the like? I don't use this functionality, so I wasn't positive

Reply:
According to @fsouza's blog post (https://blog.fsouza.dev/prettierd-neovim-format-on-save/) prettier takes about 350ms. So I guess changing the default `timeout_ms` to either 500ms or 1000ms should be plenty.

Reply:
I would test, but I imagine that 500ms is fine, as I guarantee there are people who are going to be using this with prettier.

Reply:
:+1: for increasing the default to 500ms

Reply:
Will test today with prettier and other slower formatters to see if 500 or 750 are sane defaults.